apis/v1alpha1/validation.go

```diff
@@ -57,3 +66,39 @@ func ValidateIAMRequest(r *IAMRequest) (retErr error) {
 	}
 	return
 }
+
+// ValidateCLIRequest checks if the CLIRequest is valid.
+func ValidateCLIRequest(r *CLIRequest) (retErr error) {
+	// Set default CLI
+	if r.CLI == "" {
+		r.CLI = defaultCLI
+	}
+	// TODO (#49): support other CLIs.
+	if r.CLI != defaultCLI {
+		retErr = errors.Join(retErr, fmt.Errorf("CLI %q is not supported", r.CLI))
+	}
+
+	// Check if the do commands are valid.
+	for _, c := range r.Do {
+		if err := checkCommand(c); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("command %q is not valid: %w", c, err))
+		}
+	}
+
+	// Check if the cleanup commands are valid.
+	for _, c := range r.Cleanup {
+		if err := checkCommand(c); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("command %q is not valid: %w", c, err))
+		}
+	}
+	return retErr
+}
+
+func checkCommand(c string) error {
+	for _, ch := range c {
+		if _, ok := invalidCommandOperators[ch]; ok {
+			return fmt.Errorf("invalid command operator %q", ch)
+		}
```

Comment:
Currently, if multiple invalid command operators are matched, users have to change the command multiple time as each time the validation only returns one error related to one matched invalid operator

Reply:
Right, can put the whole list in the error message. updating.

Reply:
Since the list is short, we can stop early and provide all the invalid operators.